### PR TITLE
Add eslint prettier husky and editorlog configuration for the codebase

### PR DESCRIPTION
**Pull request by DamengRandom**
* Added eslint configurations
* Added Prettier configurations
* Added Husky configurations
* Added editorlog configurations

### CODE REVIEW

File: spi-sample-pos-2/package.json [hunk 1]
```diff
@@ -20,7 +20,18 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "prepare": "cd .. && husky install spi-sample-pos-2/.husky",
+    "check-prettier": "npx prettier --check .",
```
Comment (mattgresham) [inline spi-sample-pos-2/package.json:25]: With this one, you can just do the following as `npx` will reach out remotely to grab a script to run
```suggestion
    "check-prettier": "prettier --check .",
```

**Review by mattgresham — APPROVED** [review]
One small thing that's not needed at the moment, but if you wanted to, you can move everything in the spi-sample-pos-2 folder into the directory above it. Otherwise, it can be done later at any time.

Nice work though!

File: spi-sample-pos-2/package.json (hunk 1 above)
Comment (DamengRandom) [reply spi-sample-pos-2/package.json:25]: Yeah sure, Matt, we will update this line 🙌

**Comment by DamengRandom** [comment]
HI @mattgresham, we will move the code into spi-sample-pos-2 into existing_pos directory.

**Pull request merged**